lib/executor.js

```diff
@@ -32,6 +32,7 @@ module.exports = function execute(app, instructions, callback) {
   setupModels(app, instructions);
   setupMiddleware(app, instructions);
   setupComponents(app, instructions);
+  setEnv(app, instructions);
```

Comment:
This should be the first operation executed, just before `setHost`. That way other setup procedures (e.g. initialisation of 3rd party components) see the correct `app.get('env')` value.